lucene/sandbox/src/java/org/apache/lucene/document/ShapeField.java

```diff
@@ -101,21 +143,95 @@ protected void setTriangleValue(int aX, int aY, boolean abFromShape, int bX, int
   private static final int MAXY_MINX_MINY_X_Y_MAXX = 6;
   private static final int MINY_MINX_Y_MAXX_MAXY_X = 7;
```

Comment:
What would be the pros/cons vs. adding new triangle types, e.g. as below?

```
private static final int POINT_Y_X = 8;
private static final int LINE_MAXY_MINX_MINY_MAXX = 9;
private static final int LINE_MINY_MINX_MAXY_MAXX = 10;
```

Reply:
I missed this update. The problem here is that we are using the first three bits to encode the type of triangle so this will make the change not backwards compatible although the benefit is that you only use one bit for adding the extra information.

Reply:
I guess we can get the version of the segment and act accordingly.

Reply:
Indeed this change needs to be backward compatible, but is it really an issue here? My understanding is that with all segments you would get points reported as triangles, but in practice that wouldn't be an issue? And new indices with this change should never get queried with old versions of Lucene that don't know about these constants (Lucene doesn't guarantee any forward compatibility) so that wouldn't be an issue either?